airflow/models/dag.py

```diff
@@ -1384,7 +1390,8 @@ def sync_to_db(self, owner=None, sync_time=None, session=None):
         orm_dag.default_view = self._default_view
         orm_dag.description = self.description
         orm_dag.schedule_interval = self.schedule_interval
-        session.merge(orm_dag)
+        orm_dag.tags = self.get_dagtags(session=session)
```

Comment:
I wonder if it would be better to use a JSON column on the Dag table, rather than needing a new dag-tag table.

SQLAlchemy has built in support for JSON columns for mysql, sqlite and postgres, and with postgres we could even add an index on it.

Reply:
+1  on using JSON column. Don't mind if it is in same table or different. Maybe makes sense to have it in the same Dag table just for consistency and won't need a new class DagTag :D

Reply:
@ashb @kaxil As far as I know, sqlalchemy doesn't fully support JSON Arrays, it's not possible to check if value X is in the list.
Am I missing something?

Reply:
Oh. It's possible, but we have to handle the DB differences our selves :( https://stackoverflow.com/questions/39470239/sqlalchemy-json-column-how-to-preform-a-contains-query

Reply:
Yep, it doesn't sounds best practice.
WDYT?

Reply:
Why does it need JSON column?
If table is on DAG level... each row represent dag then labels are just dimension on this DAG. It can be array type.
Each change in label affect only 1 DAG so it's just updating one cell

Reply:
@OmerJog ARRAY type is currently working only with PostgreSQL.